python/route_guide/README.md

```diff
@@ -0,0 +1,281 @@
+          yield prev_note
+      prev_notes.append(new_note)
+```
+
+This method's semantics are a combination of those of the request-streaming method and the response-streaming method. It is passed an iterator of request values and is itself an iterator of response values.
+
+### Starting the server
+
+Once you have implemented all the `RouteGuide` methods you'll likely want to start up a gRPC server so that clients can actually use your service:
+
+```python
+def serve():
+  server = route_guide_pb2.early_adopter_create_RouteGuide_server(
+      RouteGuideServicer(), 50051, None, None)
+  server.start()
+```
+
+Because `start()` does not block you may need to sleep-loop if there is nothing else for your code to do while serving.
+
+<a name="client"></a>
+## Creating the client
+
+You can see the complete example client code in [grpc-common/python/route_guide/route_guide_client.py](https://github.com/grpc/grpc-common/blob/master/python/route_guide/route_guide_client.py).
+
+### Creating a stub
+
+To call service methods, we first need to create a *stub*.
+
+We use the `early_adopter_create_RouteGuide_stub` function of the `route_guide_pb2` module generated from our .proto.
+
+```python
+stub = RouteGuide::Stub.new('localhost', 50051)
+```
+
+The returned object implements all the methods defined by the `EarlyAdopterRouteGuideStub` interface and is also a [context manager](https://docs.python.org/2/library/stdtypes.html#typecontextmanager). All RPCs invoked on the stub must be invoked within the stub's context so it is common for stubs to be created and used with a [with statement](https://docs.python.org/2/reference/compound_stmts.html#the-with-statement):
+
+```python
+with route_guide_pb2.early_adopter_create_RouteGuide_stub('localhost', 50051) as stub:
+```
+
+### Calling service methods
+
+For RPC methods that return a single response ("response-unary" methods), gRPC Python supports both synchronous and asynchronous control flow semantics. For response-streaming RPC methods, calls immediately return an iterator of response values and calls to that iterator's `next()` method block until the response to be yielded from it iterator becomes available.
+
+A synchronous call to the bidirectionally-unary `GetFeature` is nearly as straightforward as calling a local method:
+
+```python
+feature = stub.GetFeature(point, timeout_in_seconds)
+```
+
+An asynchronous call to `GetFeature` is nearly as straightforward as calling a local method asynchronously in a thread pool:
+
+```python
+feature_future = stub.GetFeature.async(point, timeout_in_seconds)
+feature = feature_future.result()
+```
+
+Calling the response-streaming `ListFeatures` is similar to working with sequence types:
+
+```python
+for feature in stub.ListFeatures(rectangle, timeout_in_seconds):
+```
+
+Calling the request-streaming `RecordRoute` is similar to passing a sequence to a local method and may be done synchronously or asynchronously:
```

Comment:
Calling the request-streaming method `RecordRoute` is similar to passing a sequence to a local method. Like the simple RPC above that also returns a single response, it can be called synchronously or asynchronously: